### PR TITLE
[accelerator][BugFix] Resolve some test for 1 gpu

### PR DESCRIPTION
**Pull request by tchaton**
## What does this PR do?

Fixes # (issue) <- this [links related issue to this PR](https://docs.github.com/en/free-pro-team@latest/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword)

## Before submitting
- [ ] Was this discussed/approved via a GitHub issue? (not for typos and docs)
- [ ] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [ ] Did you make sure your PR does only one thing, instead of bundling different changes together?
- [ ] Did you make sure to update the documentation with your changes? (if necessary)
- [ ] Did you write any new necessary tests? (not for typos and docs)
- [ ] Did you verify new and existing tests pass locally with your changes?
- [ ] Did you update the [CHANGELOG](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/CHANGELOG.md)? (not for typos, docs, test updates, or internal minor changes/refactorings)

## PR review
Anyone in the community is free to review the PR once the tests have passed.
Before you start reviewing make sure you have read [Review guidelines](https://github.com/PyTorchLightning/pytorch-lightning/wiki/Review-guidelines). In short, see the following bullet-list:

 - [ ] Is this pull request ready for review? (if not, please submit in draft mode)
 - [ ] Check that all items from **Before submitting** are resolved
 - [ ] Make sure the title is self-explanatory and the description concisely explains the PR
 - [ ] Add labels and milestones (and optionally projects) to the PR so it can be classified
 - [ ] **Check that target branch and milestone match!**

## Did you have fun?
Make sure you had fun coding 🙃

### CODE REVIEW

File: pytorch_lightning/accelerators/accelerator_connector.py [hunk 1]
```diff
@@ -481,7 +482,7 @@ def set_distributed_mode(self):
         # for DDP overwrite nb processes by requested GPUs
         if (
             self._device_type == DeviceType.GPU
-            and self._distrib_type in (DistributedType.DDP, DistributedType.DDP_SPAWN, DistributedType.DDP2)
+            and self._distrib_type in (DistributedType.DDP, DistributedType.DDP_SPAWN)
```
Comment (justusschock) [inline pytorch_lightning/accelerators/accelerator_connector.py:485]: What about DDP2?

**Review by justusschock — APPROVED** [review]

**Review by justusschock — APPROVED** [review]

File: pytorch_lightning/accelerators/accelerator_connector.py [hunk 2]
```diff
@@ -148,6 +147,7 @@ def __init__(
 
     def handle_given_plugins(self, plugins: Optional[Sequence]):
         if plugins is None:
+            self._cluster_environment = self.select_cluster_environment()
```
Comment (SeanNaren) [inline pytorch_lightning/accelerators/accelerator_connector.py:150]: Instead of this you could just initialize an empty list:

```python
plugins = plugins if plugins is not None else []
```

This way you don't need to have the return or this additional cluster environment line

File: pytorch_lightning/accelerators/accelerator_connector.py [hunk 3]
```diff
@@ -206,6 +213,8 @@ def training_type_plugin(self) -> TrainingTypePlugin:
             self._training_type_plugin = self.select_training_type_plugin()
         else:
             self._training_type_plugin = self.resolve_training_type_plugin(self._training_type_plugin)
+        # attach local_rank
+        self._training_type_plugin.task_idx = self.local_rank
```
Comment (SeanNaren) [inline pytorch_lightning/accelerators/accelerator_connector.py:217]: I hope this is just for debugging sake, because this is a bit elusive

File: pytorch_lightning/plugins/precision/apex_amp.py [hunk 4]
```diff
@@ -38,6 +38,8 @@ def connect(self, model: torch.nn.Module, optimizers, lr_schedulers):
         """Connects the precision plugin to the training process,
         configures apex and reinits the schedulers
         """
+        if model.device.type != "cuda":
+            return model, optimizers, lr_schedulers
```
Comment (SeanNaren) [inline pytorch_lightning/plugins/precision/apex_amp.py:42]: We should never get here if the device type is CUDA right?

Comment (justusschock) [reply pytorch_lightning/plugins/precision/apex_amp.py:42]: this is only needed, because in Spawn the main process (which isn't the local rank 0 here) actually calls setup as well, but without a model being on gpu, meaning that we shouldn't do anythin then.
We cannot move it to gpu before, since we then occupy memory on GPU0 in the main proc and the actual rank 0.

File: pytorch_lightning/accelerators/accelerator_connector.py (hunk 2 above)
Comment (tchaton) [reply pytorch_lightning/accelerators/accelerator_connector.py:150]: Yes.

File: pytorch_lightning/accelerators/accelerator_connector.py (hunk 3 above)
Comment (tchaton) [reply pytorch_lightning/accelerators/accelerator_connector.py:217]: Yes, and it is wrong. Let me move it in ddp_spawn

**Review by awaelchli — APPROVED** [review]
